### DISCUSSION TITLE
Welcome to hyperrogue Discussions!

### DISCUSSION THREAD

**Discussion by zenorogue** [zenorogue/hyperrogue#373] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Brensom** [comment]
please take on your support packages in Archlinux:
https://aur.archlinux.org/packages/hyperrogue
https://aur.archlinux.org/packages/hyperrogue-git

**Comment by wlatendresse** [comment]
@zenorogue 
Are suggestions for improving the readability of the codebase also welcome?

Please don't misunderstand. I'm aware that this project has grown over many years. I believe that's one of the reasons why this otherwise very promising game idea has, in my opinion, far too few followers here.

I've tried to work my way through the existing codebase myself. I think it could be made much easier for people who want to contribute, in terms of readability and extensibility.

My basic idea is to first compile all the issues that currently make it too difficult for people to contribute. And perhaps also to disclose some design explanations. Is that okay?

**Reply by zenorogue** [reply]
I think your suggestions (find out issues that make things difficult, and disclose design explanations) are definitely welcome. There is definitely lots of missing or outdated documentation. I do not want to work on updating that myself because (a) I prefer to use that time to implement new features, (b) it is not clear to me what is difficult to understand for others.

I am very happy about the existing contributors, though, and it is not clear to me whether making things easier to understand would indeed lead to a significant increase in contributions. Maybe the potential contributors assume that working on a non-Euclidean game is too difficult, or they do not like the idea of contributing their work for free when I earn money from the paid versions. I have seen some people saying it is easy to work on, and some people saying it is hard; so it is subjective, depending on what you are used to and what you are trying to achieve.

**Reply by wlatendresse** [reply]
@zenorogue 
Well, judging by the number of people involved in this project, it doesn’t exactly look like another Minecraft in the making.

And yet, the concept is intriguing enough that it could very well become exactly that.
If I were the project owner, I’d wonder why.

I have quite a few thoughts on this — not just because of my decades of development experience.
If you’d like to hear them, feel free to ask.

**Reply by wlatendresse** [reply]
@zenorogue 

There is an **Orb of Beauty**? Could it be applied to roughly 70 percent of the global population?

No, all joking aside. My astonishment at this probably arises from one of the following two facts:

1. I haven't gotten to know the game well enough yet.
2. The amount of documentation in an easily digestible format is insufficient.

Since every programmer is likely acquainted with the annoyances of documentation, particularly the inconsistencies it creates between code and documentation, I'm pondering whether it would be wise to automatically generate documentation for existing features.

It goes without saying that in order for this to function, the codebase would probably require a structure that is somewhat more hierarchical.

But perhaps this approach would also contradict the "spirit of discovery" inherent in the game?

**Reply by wlatendresse** [reply]
I recently learned that the "World Overview," which can be accessed by pressing the "o" key, offers far more in-depth details about the items and terrain.

This fact, in my opinion, need to be made far more apparent from the beginning of the program.

It's also very cleverly concealed that the **game has a wiki**.
